script/subtitles/subtitles.js

```diff
@@ -0,0 +1,79 @@
+define('bigscreenplayer/subtitles/subtitles',
+  [
+    'bigscreenplayer/subtitles/legacysubtitles',
+    'bigscreenplayer/utils/loadurl',
+    'bigscreenplayer/debugger/debugtool',
+    'bigscreenplayer/plugins'
+  ],
+  function (SubtitlesContainer, LoadURL, DebugTool, Plugins) {
+    'use strict';
+    // playbackStrategy, captionsURL, isSubtitlesEnabled(), playbackElement TODO: change the ordering of this, doesn't make sense.
+    return function (mediaPlayer, url, autoStart, playbackElement) {
+      var subtitlesContainer;
+      var subtitlesEnabled = autoStart;
+      var subtitlesAvailable = !!url;
+
+      DebugTool.info('Loading subtitles from: ' + url);
+      LoadURL(url, {
+        onLoad: function (responseXML, responseText, status) {
+          if (!responseXML) {
+            DebugTool.info('Error: responseXML is invalid.');
+            Plugins.interface.onSubtitlesTransformError();
+            return;
+          }
+
+          if (status === 200) {
+            subtitlesContainer = SubtitlesContainer(mediaPlayer, responseXML, autoStart, playbackElement);
+          }
+        },
+        onError: function (error) {
+          DebugTool.info('Error loading subtitles data: ' + error);
+          Plugins.interface.onSubtitlesLoadError();
+        }
+      });
+
+      function start () {
+        subtitlesEnabled = true;
+        if (available() && subtitlesContainer) {
+          subtitlesContainer.start();
+        }
+      }
+
+      function stop () {
+        subtitlesEnabled = false;
+        if (available() && subtitlesContainer) {
+          subtitlesContainer.stop();
```

Comment:
Would it be possible to setting the enabled state and stopping and starting the subtitles in separate functions? this would allow you to have subtitles enabled, but hidden (eg in a resized window)

Reply:
Agreed to do this in a later iteration